### PR TITLE
User dash layout and component integration

### PR DESCRIPTION
**Pull request by ialej001**
Changes include:

- Updated the Sign In Button on the boilerplate page to route to `/login`
- After login, automatically reroutes to `/dash`
- Log out redirects to `/`
- Implemented the layout for the user dashboard. Most components are bare-bones/placeholders to be iterated on.
- Integrated all previous merged branches and included `Header` in the dash
- `/dash` has some basic responsiveness.

### CODE REVIEW

File: src/components/Event/EventItem.tsx [hunk 1]
```diff
@@ -1,118 +1,153 @@
...
+          <div className="flex flex-wrap gap-1">
+            <div className="h-10 w-10 rounded-full border bg-black">
+              <div className="mt-2 text-center text-white">
+                +{props.event.eventParticipants.length}
               </div>
-              {props.event.eventParticipants.map((participants) => {
+            </div>
+            {props.event.eventParticipants.map((participant) => {
                 return (
                   <>
```
Comment (mpark4656) [inline src/components/Event/EventItem.tsx:76]: The usage of Array.map here requires the child component to have a key attribute.

File: src/components/Event/SuggestedEvents.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { EventList } from "./EventList";
+
+interface SuggestedEventsProps {
+  title: string;
+  events: EventItemType[] | EventWithParticipants[];
+}
+
+const SuggestedEvents = ({ title, events }: SuggestedEventsProps) => {
+  const isViewAllHidden = events.length > 0;
+  console.log(isViewAllHidden);
```
Comment (mpark4656) [inline src/components/Event/SuggestedEvents.tsx:13]: The debugging output could be removed if the debugging is complete.

**Review by mpark4656 — CHANGES_REQUESTED** [review]
This looks great! Just a couple of changes I added - please refer to inline comments.

File: src/components/Event/NewEvent.tsx [hunk 3]
```diff
@@ -1,11 +1,51 @@
...
+
+interface TextInputProps {
+  inputType: "text" | "textArea";
+  label: String;
+  id: string;
 }
 
-export default NewEvent
\ No newline at end of file
+const TextInput = (props: TextInputProps) => {
```
Comment (Briantam0422) [inline src/components/Event/NewEvent.tsx:29]: Should we make it a component so we can use it in other forms?

File: src/components/Event/EventItem.tsx (hunk 1 above)
Comment (ialej001) [reply src/components/Event/EventItem.tsx:76]: That is an error from the previous merge. I fixed here.

File: src/components/Event/SuggestedEvents.tsx (hunk 2 above)
Comment (ialej001) [reply src/components/Event/SuggestedEvents.tsx:13]: removed

File: src/components/Event/NewEvent.tsx (hunk 3 above)
Comment (ialej001) [reply src/components/Event/NewEvent.tsx:29]: Yeah, I'll pull it out.

**Review by Briantam0422 — APPROVED** [review]

**Review by mpark4656 — APPROVED** [review]

**Pull request merged**